doc/site/articles/netcode-overview.markdown

```diff
@@ -0,0 +1,56 @@
+
+### Ping and the packet queue
+Of course, packets do not arrive at the server immediately, your ping is involved.
+But that is not all: **to guard against network instability**, they are not broadcast by the server to other clients immediately. Instead, **packets are queued for a bit later in a buffer**.
+By default this is about about 3-6 simulation frames (100-200ms) on top of your "normal" ping, though **games can adjust it and the engine also adjusts it on its own if a match is "laggy"**.
+In interfaces such as the `/info` playerlist and `Spring.GetPlayerInfo`, the **ping is in terms of sim frames after taking the queue into account**.
+
+### Sync
+What makes the "simulate inputs" approach work is that **the engine takes utmost care to keep calculations identical** on each client.
+This is not trivial because you still have to work with **things that naturally differ on each client**, such as mouse position or which units are selected - this is **called the unsynced state**.
+On top of that, there can be **hardware differences** that have to be worked around to get identical results - the huge effort involved is one of the reasons why **Recoil is not available outside x86-64**.
+Finally, **desync can cascade in a sort of butterfly effect**.
+For example, maybe a unit deals a bit less damage than it should, which makes its target survive an otherwise lethal hit;
+that unit then bumps into a constructor, slightly delaying the construction of a resource collector.
+A difference in stored resources then stems from an incorrect damage calculation minutes earlier.
+The **huge effort of hunting down sources of desync** is one of the biggest drawbacks of the architecture.
+
+### Replays and saves
+A **replay is just a replication of the network queue from the live game**.
+Instead of receiving packets from the server, you read them from your local replay file. Otherwise **everything proceeds largely the same as a live game**.
+
+A save is an **imperfect snapshot** of the gamestate.
+While doable, a **perfect replication is difficult and not currently implemented**.
+Running identical inputs on a save will proceed differently compared to running them on the original state from which the save was produced.
+This means that **using the save mechanism is not currently possible to get mid-game joins or skipping backwards in replays** since it would desync, although could happen if somebody contributes the effort.
+
+### Tradeoffs involved
+The **benefits** of this setup are:
+* extrmely small bandwidth use (remember Recoil has its roots in Total Annihilation back from 1997)
+* small replay size
+* you cannot easily cheat by modifying your own gamestate (cheat engine style), since there are no inputs that will replicate that on the server.
+* the server doesn't need to simulate a game, since clients already do. This makes hosting much easier. See [the dedicated server article]({{ site.baseurl }}{% link guides/headless-and-dedi.markdown %}).
```

Comment:
I'd check if this link works running the site locally. I'm not sure

Reply:
exact copy-paste of this one, which works: https://github.com/beyond-all-reason/spring/blob/533f2e0a07380558058f7d04ef31ea38ce2f7d3c/doc/site/articles/wupget-communication.markdown?plain=1#L81